enhancements/installer/aws-lb-subnet-selection.md

```diff
@@ -0,0 +1,1029 @@
+
+##### Installer Validation Rules
+
+These validation rules only apply to the new `platform.aws.vpc.subnets` field.
+
+###### All or Nothing Subnet Roles Selection
+
+This proposed API follows an "all or nothing" approach for roles: either all roles must be explicitly assigned
+(manual role selection), or no roles must be assigned (automatic role selection). The installer must not allow a mix of
+automatic and manual role assignments, as this could lead to confusing behavior. See [Allowing Mixed Automatic and Manual Subnet Selection Alternative](#allowing-mixed-automatic-and-manual-subnet-selection-alternative) for the reason behind this validation.
+
+However, one exception is if the cluster is private (`installconfig.publish: Internal`), then
+ControlPlaneExternalLB can be omitted.
+
+###### All Subnets Belong to the Same VPC Validation
+
+The [existing validation](https://github.com/openshift/installer/blob/0d77aa8df5ddc68e926aa11da24a87981021b256/pkg/asset/installconfig/aws/subnet.go#L91)
+for the deprecated `platform.aws.subnets` field ensuring all subnets belong to the same VPC
+should apply to all subnets specified in `platform.aws.vpc.subnets`.
+
+###### Multiple Load Balancer Subnets in the Same AZ Validation
+
+The installer must reject multiple `IngressControllerLB` role subnets in the same AZ as this will be
+rejected by the AWS CCM.
+
+###### The Old and New Subnets Fields Cannot be Specified Together Validation
+
+Since `platform.aws.subnets` will be deprecated, the installer must not allow `platform.aws.vpc.subnets` and
+`platform.aws.subnets` to both be specified.
+
+###### Maximum of 10 IngressController Subnets Validation
+
+Since the IngressController's API [only allows 10 subnets](https://github.com/openshift/api/blob/ee5bb7eaf6b6638d4e3b33ba4ff0834212cdb75d/operator/v1/types_ingress.go#L564),
+the installer must not allow more than 10 IngressController subnets as well.
+
+###### Consistent Cluster Scope with IngressControllerLB Subnets Validation
+
+The installer must not allow **any** public IngressControllerLB subnets for private clusters (`installconfig.publish: Internal`)
+as this is a security risk as discussed in the [Do Not Use Untagged Subnets User Stories](#do-not-use-untagged-subnets-user-stories) section.
+
+Conversely, the installer must now allow **any** private IngressControllerLB subnets for public clusters
+(`installconfig.publish: External`) as the public IngressController will not function with private subnets.
+
+###### Consistent Cluster Scope with ControlPlaneLB Subnets Validation
+
+The installer must not allow **any** public subnets to be specified with the ControlPlaneInternalLB role and must
+not allow **any** private subnets to be specified with the ControlPlaneExternalLB role.
+
+Private clusters (`installconfig.publish: Internal`) should reject an install-config containing
+any ControlPlaneExternalLB roles, as only an internal control plane load balancer will be created.
```

Comment:
Piggybacking on my earlier comment, I think we can avoid this validation and improve the user experience, by not separating `ControlPlaneExternalLB` and `ControlPlaneInternalLB` into separate values, but just have a `ControlPlaneLB` value and select the appropriate LB based on whether the subnet is public/private.

Reply:
In this case, what would happen if all of the subnets were private? What would happen if they were all public? 

Is it not more obvious (from an early validation perspective) to require these separately, if we know that some should be public and some should be private?

Reply:
Holding off on combining ControlPlaneExternalLB/ControlPlaneInternalLB into ControlPlaneLB for now. Something we can discuss Monday.